src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -4631,6 +4608,295 @@ impl Coordinator {
+
+            let internal_cmd_tx = self.internal_cmd_tx.clone();
+            let transient_revision = self.catalog().transient_revision();
+            let conn_id = ctx.session().conn_id().clone();
+            let connection_context = self.connection_context.clone();
+            let otel_ctx = OpenTelemetryContext::obtain();
+            let role_metadata = ctx.session().role_metadata().clone();
+
+            task::spawn(
+                || format!("validate_alter_connection:{conn_id}"),
+                async move {
+                    let dependency_ids = conn.resolved_ids.0.clone();
+                    let result = match connection.validate(id, &connection_context).await {
+                        Ok(()) => Ok(conn),
+                        Err(err) => Err(err.into()),
+                    };
+
+                    // It is not an error for validation to complete after `internal_cmd_rx` is dropped.
+                    let result = internal_cmd_tx.send(Message::AlterConnectionValidationReady(
+                        AlterConnectionValidationReady {
+                            ctx,
+                            result,
+                            connection_gid: id,
+                            plan_validity: PlanValidity {
+                                transient_revision,
+                                dependency_ids,
+                                cluster_id: None,
+                                replica_id: None,
+                                role_metadata,
+                            },
+                            otel_ctx,
+                        },
+                    ));
+                    if let Err(e) = result {
+                        tracing::warn!("internal_cmd_rx dropped before we could send: {:?}", e);
+                    }
+                },
+            );
+        } else {
+            let result = self
+                .sequence_alter_connection_stage_finish(ctx.session_mut(), id, conn)
+                .await;
+            ctx.retire(result);
+        }
+    }
+
+    #[tracing::instrument(level = "debug", skip(self))]
+    pub(crate) async fn sequence_alter_connection_stage_finish(
+        &mut self,
+        session: &mut Session,
+        id: GlobalId,
+        mut connection: catalog::Connection,
+    ) -> Result<ExecuteResponse, AdapterError> {
+        match &mut connection.connection {
+            mz_storage_types::connections::Connection::AwsPrivatelink(ref privatelink) => {
+                let spec = VpcEndpointConfig {
+                    aws_service_name: privatelink.service_name.to_owned(),
+                    availability_zone_ids: privatelink.availability_zones.to_owned(),
+                };
+                self.cloud_resource_controller
+                    .as_ref()
+                    .ok_or(AdapterError::Unsupported("AWS PrivateLink connections"))?
+                    .ensure_vpc_endpoint(id, spec)
+                    .await?;
```

Comment:
Should we do this after the catalog transaction so we don't leave the resource?

Reply:
Good catch. Fixed locally and will be resolved in my next push.